File: p2/deepspace/src/deepspace/SuppliesPackage.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package deepspace;

/**
 *
 * @author cristobalmer
 */
class SuppliesPackage {
    private float ammoPower;
    private float fuelUnits;
    private float shieldPower;
    
    SuppliesPackage(float _ammoPower,float _fuelUnits,float _shieldPower){
        ammoPower = _ammoPower;
        fuelUnits = _fuelUnits;
        shieldPower = _shieldPower;
    }
    SuppliesPackage(SuppliesPackage s){
        ammoPower = s.ammoPower;
        fuelUnits = s.fuelUnits;
        shieldPower = s.shieldPower;
    }
    public float getAmmoPower(){
        return ammoPower;
    }    
    public float getFuelUnits(){
        return fuelUnits;
    }    
    public float getShieldPower(){
        return shieldPower;
    }    
    
    @Override
    public String toString(){
        String result= new String("Suppliespackage with ammoPower: ");
        result+=String.valueOf(ammoPower)+"\nfuelUnits: "+String.valueOf(fuelUnits)+"\nshieldPower: "+String.valueOf(shieldPower);
        return result;
    }
}

File: p3anopasado/SuppliesPackage.rb
#encoding:utf-8
module Deepspace 
	class SuppliesPackage
		def initialize(_ammoPower, _fuelUnits, _shieldPower)
			@ammoPower=_ammoPower
			@fuelUnits=_fuelUnits
			@shieldPower=_shieldPower
		end
		
		def ammoPower()
			@ammoPower
		end
		
		def fuelUnits()
			@fuelUnits
		end
		
		def shieldPower()
			@shieldPower
		end

		def self.newCopy(s)
			
			SuppliesPackage.new(s.ammoPower, s.fuelUnits, s.shieldPower)
		end
		
		
	end
end


